modyn/storage/src/internal/file_watcher/file_watcher.cpp

```diff
@@ -0,0 +1,339 @@
+#include "internal/file_watcher/file_watcher.hpp"
+
+#include <fmt/format.h>
+#include <spdlog/spdlog.h>
+
+#include <csignal>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+using namespace storage;
+
+/*
+ * Handles the file paths that are passsed.
+ *
+ * Checks if the file is valid and if so, inserts the file into the database.
+ *
+ * Valid files are files that pass the checks in check_valid_file().
+ *
+ * @param file_paths The file paths to be handled.
+ * @param data_file_extension The extension of the data files.
+ * @param file_wrapper_type The type of the file wrapper.
+ * @param timestamp The timestamp to be used for the file.
+ */
+void FileWatcher::handle_file_paths(const std::vector<std::string>& file_paths,  // NOLINT (misc-unused-parameters)
+                                    const std::string& data_file_extension, const FileWrapperType& file_wrapper_type,
+                                    int64_t timestamp, const YAML::Node& file_wrapper_config) {
+  soci::session session = storage_database_connection_.get_session();
+
+  std::deque<std::string> valid_files;
+  for (const auto& file_path : file_paths) {
+    if (check_valid_file(file_path, data_file_extension, /*ignore_last_timestamp=*/false, timestamp)) {
+      valid_files.push_back(file_path);
+    }
+  }
+
+  SPDLOG_INFO("Found {} valid files", valid_files.size());
+
+  if (!valid_files.empty()) {
+    std::string file_path = valid_files.front();
+    int64_t number_of_samples;
+    std::vector<std::tuple<int64_t, int64_t, int32_t, int32_t>> file_frame =
+        std::vector<std::tuple<int64_t, int64_t, int32_t, int32_t>>();
```

Comment:
I don't see the purpose of this. Should I add a comment where it is defined? Or how?

Where seems redundant imo (most IDEs nowadays provide shift + click) and how would lead to issues as well (changing implementation).